### PR TITLE
Don't cache falsey values for proxy checks

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

Negative responses from the proxy utils are generally usually temporary, and caching them doesn't do much for us but create issues (e.g. #3179, #3460).  This changes these proxy utils to use a modified form of lodash's `memoize()` to prevent caching of falsey values.  This should reduce the chances of clients repeatedly thinking they don't have a proxy.

Also, I originally was under the impression that `memoize` would cache error states but I was unable to reproduce that behavior in tests, so that's not considered here.

### Checklist:

- [ ] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by mikeshultz** [comment]
@nick would appreciate your review on this since you've been critical of custom caching in the past.  I really think this is the only decent way forward for proxy data because the current reset cache functionality has been plagued by race conditions and other failures.

**Comment by mikeshultz** [comment]
Will add tests. :+1:

**Comment by mikeshultz** [comment]
Adding tests revealed a couple of issues.  Looks good now though.  :heavy_check_mark:

File: packages/graphql/src/utils/memorize.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * memorize is essentially memoize but only caches truthy values.
+ *
+ * @param {function} the function to wrap
+ * @param {function} function to use to derive the cache key
+ * @returns {function} the wrapped funtion
+ */
+export default function memorize(func, resolver) {
+  if (
+    typeof func != 'function' ||
```
Comment (nick) [inline packages/graphql/src/utils/memorize.js]: `!==`

Comment (mikeshultz) [reply packages/graphql/src/utils/memorize.js]: Blame lodash!  will change though.  :+1:

**Pull request merged**